### PR TITLE
Add an option parser and a dependencies.yaml formatter

### PR DESCRIPTION
**Pull request by johnynek**
finally adds an option parser. This breaks previous users: they have to update to the new argument format.

you can use the parse_project tool to generate or format the dependencies file.

This also serves as a demonstration case of https://github.com/johnynek/paiges

cc @ianoc @non

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -1,85 +1,86 @@
...
   io.circe:
     circe:
-      version: "0.5.0-M2"
-      modules: [core, jackson, parser]
       lang: scala
-    circe-generic:
+      modules: [ "core", "jackson", "parser" ]
       version: "0.5.0-M2"
+    circe-generic:
+      exports: [ "com.chuusai:shapeless", "org.typelevel:cats-core", "org.typelevel:cats-kernel",
```
Comment (ianoc-stripe) [inline dependencies.yaml]: The other format for the list exports is probably nicer is it?

File: dependencies.yaml [hunk 2]
```diff
@@ -1,85 +1,86 @@
...
-
-  org.scala-lang.modules:
-    scala-xml:
-      lang: scala
-      version: "1.0.6"
-
 replacements:
   org.scala-lang:
     scala-library:
-      lang: scala/unmangled # scala-library is not mangled like sbt does with other jars
```
Comment (ianoc-stripe) [inline dependencies.yaml]: loosing comments?

File: dependencies.yaml [hunk 3]
```diff
@@ -1,85 +1,86 @@
...
+      lang: "scala/unmangled"
```
Comment (ianoc-stripe) [inline dependencies.yaml]: interesting it added the `"` , what are the rules for that?

File: src/scala/com/github/johnynek/bazel_deps/DepsModel.scala [hunk 4]
```diff
@@ -55,7 +60,7 @@ case class Model(
     val reps = replacements.map { r => ("replacements", r.toDoc) }
     val opts = options.map { o => ("options", o.toDoc) }
 
-    yamlMap(List(opts, deps, reps).collect { case Some(kv) => kv })
+    yamlMap(List(opts, deps, reps).collect { case Some(kv) => kv }, 2)
```
Comment (ianoc-stripe) [inline src/scala/com/github/johnynek/bazel_deps/DepsModel.scala]: Could probably do with a doc for the lines arg above, not sure what the `2` here does

**Comment by ianoc-stripe** [comment]
Very neet!

File: dependencies.yaml (hunk 3 above)
Comment (johnynek) [inline dependencies.yaml]: for language I could probably omit them. I can make that change. `version` is often ambiguous with a number: `version: 1.2` which breaks things. So always using a quote for a version is a good idea. I could do something like, if it does not parse as a number don't use the quote, else do, but that seems ugly?

File: dependencies.yaml (hunk 2 above)
Comment (johnynek) [inline dependencies.yaml]: don't see how to fix that. :/ snakeyaml strips them as it goes.

File: dependencies.yaml (hunk 1 above)
Comment (johnynek) [inline dependencies.yaml]: yeah, I can switch to that.

File: dependencies.yaml (hunk 3 above)
Comment (ianoc-stripe) [inline dependencies.yaml]: I think its probably pretty peaceful to say quotes always in values, its consistent then

File: dependencies.yaml (hunk 2 above)
Comment (ianoc-stripe) [inline dependencies.yaml]: :/

**Comment by ianoc-stripe** [comment]
This all looks great, the only little bit of fud I might have is that we wind up stripping all the comments. but ¯\_(ツ)_/¯

**Pull request merged**